TideApp/TideApp/Helpers/ViewControllerResolver.swift

```diff
@@ -0,0 +1,46 @@
+//
+//  ViewControllerResolver.swift
+//  TideApp
+//
+//  Created by John Sanderson on 17/05/2021.
+//
+
+import SwiftUI
+import UIKit
+
+/// Resolves the parent view controller of a swift UI view
+final class ViewControllerResolver: UIViewControllerRepresentable {
```

Comment:
May I have more explanation on what means resolve the parent view? Sorry just curious

Reply:
This basically gives us a way to access the "Hosting view controller" of a swift ui view. Which then allows us to access properties such as the `navigationController` etc. 
<img width="464" alt="Screenshot 2021-05-18 at 14 51 25" src="https://user-images.githubusercontent.com/6672705/118663611-bc0a0100-b7e8-11eb-910c-0d89f041d2ca.png">